lib/ivis_opengl/gfx_api_gl.cpp

```diff
@@ -62,6 +92,38 @@ struct gl_texture : public gfx_api::texture
 
 };
 
+struct gl_buffer : public gfx_api::buffer
+{
+	gfx_api::buffer::usage usage;
+	GLuint buffer;
+
+	virtual ~gl_buffer() override
+	{
+		glDeleteBuffers(1, &buffer);
+	}
+
+	gl_buffer(const gfx_api::buffer::usage& usage, const size_t& size, const gfx_api::context::buffer_storage_hint& hint)
+		: usage(usage)
+	{
+		glGenBuffers(1, &buffer);
+		glBindBuffer(to_gl(usage), buffer);
+		glBufferData(to_gl(usage), size, nullptr, to_gl(hint));
+	}
+
+	void bind() override
+	{
+		glBindBuffer(to_gl(usage), buffer);
+	}
```

Comment:
Thinking about this a bit more, `bind()` ought to be `bind(const gfx_api::buffer::usage& usage)` to properly enable OpenGL best-practices.

It is very common - and very useful - to bind a buffer to different binding targets. As-is, this abstraction does not permit that at all. Instead, you'd have to create a new `gl_buffer`, which allocates new memory and copies data again, which will introduce a huge performance bottleneck.